xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/BaseModal.java

```diff
@@ -51,6 +59,9 @@ public BaseModal(By selector)
         String className = this.container.getAttribute("class");
         className = className.replace("fade", "");
         getDriver().executeScript("arguments[0].setAttribute(\"class\",arguments[1])", this.container, className);
+        // Once we're sure the modal is loaded, we can validate its content in regards to accessibility
+        className = this.container.getAttribute("class").split(" ")[0];
+        validateWCAG(getUtil().getWCAGUtils().getWCAGContext(), true, "." + className);
```

Comment:
Feels dangerous to rely on the className only: what if you have a page with 3 modals having the class `modal`, and only one is loaded? Right now the validation will be executed on the 3 modals AFAIU. 
Feels like you could artificially add a new specific class (pretty much like we remove the fade class) just to identify the actual modal.